### ISSUE TITLE
build failure with repository on UNC path.

### ISSUE THREAD

**Issue by msftrncs** [PowerShell/PowerShell#9395]
trying to build the repository, I run in to an error.  I almost exclusively store files on UNC paths.  Why?  Guaranteed there is a bug out there to be squashed when it comes to UNC paths.

Running `Start-PSBootstrap` resulted in
```None
Installing PowerShell build dependencies
WARNING: Could not find 'dotnet', appending C:\Users\Carl\AppData\Local\Microsoft\dotnet to PATH.
WARNING: Still could not find 'dotnet', restoring PATH.
dotnet not present.  Installing dotnet.
Processing -File 'Microsoft.PowerShell.Core\FileSystem::\\svr-2015-01\User Documents\Carl\Git Repositories\PowerShell\tools\dotnet-install.ps1' failed: The given path's format is not supported. Specify a valid path for the -File parameter.
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

Execution of { & $fullPSPath -NoLogo -NoProfile -File $fullDotnetInstallPath -Channel $Channel -Version $Version } by build.psm1: line 1541 failed with exit code -196608
At \\svr-2015-01\User Documents\Carl\Git Repositories\PowerShell\build.psm1:2036 char:17
+                 throw $errorMessage
+                 ~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (Execution of { & $f\u2026h exit code -196608:String) [], RuntimeException
+ FullyQualifiedErrorId : Execution of { & $fullPSPath -NoLogo -NoProfile -File $fullDotnetInstallPath -Channel $Channel -Version $Version } by build.psm1: line 1541 failed with exit code -196608
```

PowerShell doesn't even support its own path syntax for its -file parameter.    This appears that this is just a problem with Windows PowerShell, which is being called to support the dotnet installer on Windows.  It looks like PowerShell core does not exhibit this issue.

**Comment by iSazonov** [comment]
It is not clear that you want - build PowerShell Core from UNC or report an language issue/bug? For first I think it make no sense to fix build scripts until we will really need this in development process. For second please make simple repo steps to demo a problem you want fix.

**Comment by msftrncs** [comment]
Step 1:  Attempt to build a fresh clone of PowerShell Core from a UNC path in a fresh environment.

The build process fails at this point:
https://github.com/PowerShell/PowerShell/blob/9140e421eceeef9d92cf671cc6aaa0782ea343cd/build.psm1#L1541

The actual issue is that Windows PowerShell's FILE parameter does not accept 'provider path notation' and the UNC path when resolved by JOIN-PATH results in a PSPATH that contains the provider which is then sent to Windows PowerShell which fails. I am really surprised that more people do not use UNC paths, but I suppose even 'nix users utilize mount points (links) instead of UNC paths.

I relocated the clone of the repository to a local drive, and let the install-dotnet script run there, but I suspect that the install-dotnet script would have worked in a UNC path, but didn't try it.  I could try it, and if it works, I think I could suggest a solution to the install-dotnet function that would allow for calling Windows PowerShell with the install-dotnet script path.

https://github.com/PowerShell/PowerShell/blob/9140e421eceeef9d92cf671cc6aaa0782ea343cd/build.psm1#L1540

`$fullDotnetInstallPath = (Join-Path -Path $pwd.Path -ChildPath $installScript) -replace '^[^:]*::', ''`

**Comment by iSazonov** [comment]
> the install-dotnet script would have worked in a UNC path

@TravisEz13 Could you please comment this? Make sense to track/fix this?

**Comment by msftrncs** [comment]
Alternately:

`$fullDotnetInstallPath = Join-Path -Path $pwd.ProviderPath -ChildPath $installScript`

**Comment by TravisEz13** [comment]
I agree this is an issue with the script, but building from UNC would **not be officially supported** as it creates issues with debugging.  But on the other hand, most people don't run in the supported configuration.  We only check that if you go to create a package.
